File: Answer/171.excel表列序号.cs
/*
 * @lc app=leetcode.cn id=171 lang=csharp
 *
 * [171] Excel表列序号
 */

// @lc code=start
public class Solution {
    public int TitleToNumber(string s) {
        if (s.Length == 0) return 0;
        int n = 0;
        for (int i = 0; i < s.Length; i++)
        {
            n += (int)(((int)s[i] - 64) * Math.Pow(26, (s.Length - i - 1)));
        }
        return n;
    }
}
// @lc code=end

File: Answer/171.excel表列序号.cpp
/*
 * @lc app=leetcode.cn id=171 lang=cpp
 *
 * [171] Excel表列序号
 */

// @lc code=start
class Solution {
public:
    int titleToNumber(string s) {
        if(s.length() == 0) return 0;
        int n = 0;
        for(int i = 0; i < s.length(); i++)
        {
            n += ((int)s[i] - 64) * pow(26, (s.length() - i - 1));
        }
        return n;
    }
};
// @lc code=end

